contracts/IdleMStableStrategy.sol

```diff
@@ -0,0 +1,235 @@
+        vault = IVault(_vault);
+        govToken = vault.getRewardToken();
+        idleCDO = _idleCDO;
+
+        uniswapRouterPath = routerPath;
+        uniswapV2Router02 = IUniswapV2Router02(_uniswapV2Router02);
+
+        ERC20Upgradeable.__ERC20_init("Idle MStable Strategy Token", string(abi.encodePacked("idleMS", underlyingToken.symbol())));
+        lastIndexedTime = block.timestamp;
+        //------//-------//
+
+        transferOwnership(_owner);
+    }
+
+    // only claim gov token rewards
+    function redeemRewards() external onlyIdleCDO returns (uint256[] memory rewards) {
+        _claimGovernanceTokens(0, 0);
+        rewards = new uint256[](1);
+        rewards[0] = _swapGovTokenOnUniswapAndDepositToVault(0); // will redeem whatever possible reward is available
+    }
+
+    function redeemRewards(bytes calldata _extraData) external override onlyIdleCDO returns (uint256[] memory rewards) {
+        (uint256 minLiquidityTokenToReceive, uint256 startRound, uint256 endRound) = abi.decode(_extraData, (uint256, uint256, uint256));
+        _claimGovernanceTokens(startRound, endRound);
+        rewards = new uint256[](1);
+        rewards[0] = _swapGovTokenOnUniswapAndDepositToVault(minLiquidityTokenToReceive);
+    }
+
+    function pullStkAAVE() external override returns (uint256) {
+        return 0;
+    }
+
+    function price() public view override returns (uint256) {
+        return imUSD.exchangeRate();
+    }
+
+    function getRewardTokens() external view override returns (address[] memory) {
+        address[] memory govTokens;
+        govTokens[0] = govToken;
+        return govTokens;
+    }
+
+    function deposit(uint256 _amount) external override onlyIdleCDO returns (uint256 minted) {
+        require(_amount != 0, "Deposit amount should be greater than 0");
+        underlyingToken.transferFrom(msg.sender, address(this), _amount);
+        return _depositToVault(_amount);
+    }
+
+    function _depositToVault(uint256 _amount) internal returns (uint256) {
+        underlyingToken.approve(address(imUSD), _amount);
+        lastIndexAmount = lastIndexAmount + _amount;
+        lastIndexedTime = block.timestamp;
+        uint256 interestTokensReceived = imUSD.depositSavings(_amount);
+
+        uint256 interestTokenAvailable = imUSD.balanceOf(address(this));
+        imUSD.approve(address(vault), interestTokenAvailable);
+
+        vault.stake(interestTokenAvailable);
+
+        _mint(msg.sender, interestTokenAvailable);
+        return interestTokenAvailable;
+    }
+
+    // _amount is strategy token
+    function redeem(uint256 _amount) external override onlyIdleCDO returns (uint256) {
+        return _redeem(_amount);
+    }
+
+    // _amount in underlying token
+    function redeemUnderlying(uint256 _amount) external override returns (uint256) {
+        uint256 _underlyingAmount = (_amount * oneToken) / price();
+        lastIndexAmount = lastIndexAmount - _underlyingAmount;
```

Comment:
Why in `redeemUnderlying` the `lastIndexAmount` var is updated while in the `_redeem` it's not? it should be updated in both cases

Reply:
same for `lastIndexedTime`

Reply:
Changed